File: testframwork/javaandscalatest/src/main/scala/com/testframwork/jdk8/streamTest/lambdaTest/Demo02.java
package com.testframwork.jdk8.streamTest.lambdaTest;

import com.testframwork.jdk8.optionalTest.Student;
import com.testframwork.jdk8.streamTest.Test01;
import com.testframwork.jdk8.streamTest.lambdaTest.TestInterface.FunctionalInterfaceTest01;
import org.junit.Test;

import java.util.List;
import java.util.stream.Collectors;

public class Demo02 {

    private static boolean getName(Student x) {
        Test01 t = new Test01();
        List<com.testframwork.jdk8.streamTest.Student> filt = t.getStudent().stream().filter(e -> e.getName().equals("第一位")).collect(Collectors.toList());

        System.out.println(filt);
        return filt.size() == 0;
    }

    @Test
    public void test01() {
//这种方法相当于是将函数式接口进行了实现
        func01(Demo02::getName);
    }

    @Test
    public void test02() {
        func01(x ->
                x.getName().equals(null));
    }


//    private void func(FunctionalInterfaceTest interfaceParam) {
//        Student x = new Student();
//        interfaceParam.getName(x);
//    }
//
    private void func01(FunctionalInterfaceTest01 interfaceTest01) {
        Student x = new Student();
        interfaceTest01.getName(x);
    }

}

File: testframwork/javaandscalatest/src/main/scala/com/testframwork/testscala/test02/Demo02.scala
package com.testframwork.testscala.test02

object Demo02 {
  def main(args: Array[String]): Unit = {

    //定义一个数组
    val arr = Array(1, 2, 3, 4, 5)
    for (item <- arr) {

      println(item + " ")
    }

    //取出数组中的偶数元素
    for (item <- arr) {
      if (item % 2 == 0) {
        println("item:   ",item)
      }
    }
    //另外一种方法
    for(item<- arr if (item %2 ==0)){
      print("偶数：  ",item)
    }


  }


}
